refactor(dev-infra): improve error message for unexpected version branches

devversion opened the pull request:
Currently the merge script default branch configuration throws an error
if an unexpected version branch is discovered. The error right now
assumes to much knowledge of the logic and the document outlining
the release trains conceptually.

We change it to something more easy to understand that doesn't require
full understanding of the versioning/labeling/branching document that
has been created for the Angular organization.

josephperrott commented on dev-infra/pr/merge/defaults/branches.ts:
```diff
@@ -177,15 +182,21 @@ export async function findActiveVersionBranches(
-          `deleting the branch, or check if the version in "${nextBranchName}" is outdated.`);
+          `Discovered unexpected version-branch "${name}" for a release-train that is already ` +
+          `active in the "${nextBranchName}" branch. Please either delete the branch if ` +
+          `created by accident, or update the version in the next branch.`);
```
Nit: should we use the `nextBranchName` here?

```suggestion
          `created by accident, or update the version in the ${nextBranchName} branch.`);
```

devversion replied:
Good question. I wanted to use `next` branch here to partially use the terminology that matches up a bit with Pete's document (hence also the use of version-branch and release-train), but it might be rather confusing as it's not clear that "master" === "next branch". 

Will change. let me know what you think.

josephperrott replied:
maybe include both `update the version in the next branch (master)`?

devversion replied:
oh, love that!

devversion replied:
done. please have another look

josephperrott reviewed (approved):
LGTM, just one small thing to look at

devversion committed:
refactor(dev-infra): improve error message for unexpected version branches

Currently the merge script default branch configuration throws an error
if an unexpected version branch is discovered. The error right now
assumes to much knowledge of the logic and the document outlining
the release trains conceptually.

We change it to something more easy to understand that doesn't require
full understanding of the versioning/labeling/branching document that
has been created for the Angular organization.
